zkvm/src/verifier.rs

```diff
@@ -55,7 +55,7 @@ impl<'a, 'b> Delegate<r1cs::Verifier<'a, 'b>> for Verifier<'a, 'b> {
         &mut self,
         run: &mut Self::RunType,
     ) -> Result<Option<Instruction>, VMError> {
-        let mut program = Subslice::new_with_range(&run.program, run.offset..run.program.len())?;
+        let mut program = SliceReader::new_with_range(&run.program, run.offset..run.program.len())?;
```

Comment:
Per suggestion (5) in https://github.com/interstellar/slingshot/issues/80 we can use `::parse` here like so:

```rust
let (instr, remainder) SliceReader::parse(&run.program[run.offset..], |r| {
    (Instruction::parse(r)?, r.skip_trailing_bytes())
})
if remainder.len() == 0 {
    return Ok(None);
}
run.offset = run.program.len() - remainder;
Ok(Some(instr))
```